applications/tari_console_wallet/src/ui/state/app_state.rs

```diff
@@ -643,9 +665,23 @@ impl AppStateInner {
     }
 
     pub async fn refresh_balance(&mut self) -> Result<(), UiError> {
-        let balance = self.wallet.output_manager_service.get_balance().await?;
-        self.data.balance = balance;
-        self.updated = true;
+        if self.balance_enquiry_cooldown_start.elapsed() > self.balance_enquiry_cooldown_period {
```

Comment:
There is no need for an additional method here, and the names are confusing. I would suggest removing the `refresh_balance_check` method and the `balance_enquiry_cooldown_infringement` variables as calling the `refresh_balance` method twice in quick succession has the same effect

Reply:
Implemented a balance check debouncer for these purposes - please see latest commit